upgrade_through_versions_test.py

```diff
@@ -15,160 +11,49 @@
 from Queue import Empty, Full
 
 import psutil
-
+import schema_metadata_test
 from cassandra import ConsistencyLevel, WriteTimeout
 from cassandra.query import SimpleStatement
-from dtest import DEFAULT_DIR, Tester, debug
+from dtest import Tester, debug
 from tools import generate_ssl_stores, new_node
 
-# Versions are tuples of (major_ver, minor_ver)
-# Used to build upgrade path(s) for tests. Some tests will go from start to finish,
-# other tests will focus on single upgrades from UPGRADE_PATH[n] to UPGRADE_PATH[n+1]
-
-TRUNK_VER = (3, 2)
-
-# maps protocol version to c* version(s)
-PROTOCOL_PATHS = {
-    1: [(2, 0), (2, 1), (2, 2)],
-    2: [(2, 0), (2, 1), (2, 2)],
-    3: [(2, 1), (2, 2), (3, 0), (3, 1), TRUNK_VER],
-    4: [(2, 2), (3, 0), (3, 1), TRUNK_VER]
-}
-
-PROTOCOL_VERSION = int(os.environ.get('PROTOCOL_VERSION', 3))
-
-CUSTOM_PATH = os.environ.get('UPGRADE_PATH', None)
-if CUSTOM_PATH:
-    # provide a custom path like so: "1_2:2_0:2_1" to test upgrading from 1.2 to 2.0 to 2.1
-    UPGRADE_PATH = []
-    for _vertup in CUSTOM_PATH.split(':'):
-        _major, _minor = _vertup.split('_')
-        UPGRADE_PATH.append((int(_major), int(_minor)))
-else:
-    UPGRADE_PATH = PROTOCOL_PATHS[PROTOCOL_VERSION]
-
-LOCAL_MODE = os.environ.get('LOCAL_MODE', '').lower() in ('yes', 'true')
-if LOCAL_MODE:
-    REPO_LOCATION = os.environ.get('CASSANDRA_DIR')
-else:
-    REPO_LOCATION = "https://git-wip-us.apache.org/repos/asf/cassandra.git"
-
-# lets cache this once so we don't make a bunch of remote requests
-GIT_LS = subprocess.check_output(["git", "ls-remote", "-h", "-t", REPO_LOCATION]).rstrip()
-
-# maps ref type (branch, tags) to ref names and sha's
-MAPPED_REFS = defaultdict(dict)
-for row in GIT_LS.split('\n'):
-    sha, _fullref = row.split('\t')
-    _, ref_type, ref = _fullref.split('/')
-    MAPPED_REFS[ref_type][ref.split('^')[0]] = sha
-
-# We often want this post-mortem when debugging may have been disabled, so print/pprint is intentional here
-print("************************************* GIT REFS USED FOR THIS TEST RUN *********************************************")
-print("************************** KEEP IN MIND THAT A SHA MAY POINT TO ANOTHER COMMIT SHA! *******************************")
-for ref_type in MAPPED_REFS.keys():
-    print("Git refs for {}:").format(ref_type.upper())
-    pprint.pprint(MAPPED_REFS[ref_type], indent=4)
-
-if os.environ.get('CASSANDRA_VERSION'):
-    debug('CASSANDRA_VERSION is not used by upgrade tests!')
-
-
-def sha_for_ref_name(ref_name, ref_type='tags'):
-    return MAPPED_REFS[ref_type][ref_name]
+TRUNK_VER = '3.2'
+DEFAULT_PROTOCOL_VERSION = int(os.environ.get('DEFAULT_PROTOCOL_VERSION', 3))
+latest_2dot0 = 'cassandra-2.0.17'
+latest_2dot1 = 'cassandra-2.1.12'
+latest_2dot2 = 'cassandra-2.2.4'
+latest_3dot0 = 'cassandra-3.0.2'
+latest_3dot1 = 'cassandra-3.1.1'
+trunk_ver = 'git:trunk'
```

Comment:
Confusing to have a `trunk_ver` and a `TRUNK_VER`. Maybe `trunk_ver` could be `trunk_ccm_string` or something?